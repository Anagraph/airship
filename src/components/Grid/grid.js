import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';

const px = (n) =>
  typeof n === 'number' ? n + 'px' : n.indexOf('px') >= 0 ? n : n + 'px';

const width = (props) => ({
  gridTemplateColumns: `repeat(auto-fit, minmax(${px(props.width)}, 1fr))`
});

const gap = (props) => ({
  gridGap: px(props.gap)
});

const align = (props) =>
  props.align
    ? {
        alignItems: props.align
      }
    : null;

const Grid = (props) => {
  const styles = css({
    display: 'grid',
    ...width(props),
    ...gap(props),
    ...align(props)
  });

  return <div className={styles}>{props.children}</div>;
};

Grid.propTypes = {
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  gap: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  align: PropTypes.string
};

Grid.defaultProps = {
  width: 320,
  gap: 32
};

export default Grid;
