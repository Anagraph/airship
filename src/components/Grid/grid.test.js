import React from 'react';
import Grid from './Grid';
import renderer from 'react-test-renderer';

describe('render', () => {
  it('renders without crashing', () => {
    const component = renderer.create(<Grid />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with width param', () => {
    const component = renderer.create(<Grid width={300} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with gap param', () => {
    const component = renderer.create(<Grid gap={30} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with strings', () => {
    const component = renderer.create(<Grid width="300" gap="20" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
