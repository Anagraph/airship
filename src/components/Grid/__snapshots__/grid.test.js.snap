// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders with gap param 1`] = `
<div
  className="css-y7w61i"
/>
`;

exports[`render renders with strings 1`] = `
<div
  className="css-l96mez"
/>
`;

exports[`render renders with width param 1`] = `
<div
  className="css-ljyc2r"
/>
`;

exports[`render renders without crashing 1`] = `
<div
  className="css-1tvblke"
/>
`;
